tests/test_components/test_boundaries.py

```diff
@@ -188,3 +191,27 @@ def test_boundaryspec_classmethods():
     assert all(
         isinstance(boundary, PML) for boundary_dim in boundaries for boundary in boundary_dim
     )
+
+
+@pytest.mark.parametrize("absorber_type", [PML, StablePML, Absorber])
+def test_num_layers_validator(absorber_type):
+    """Test the Field validators that enforce ``num_layers>0``."""
+    with pytest.raises(pydantic.ValidationError):
+        _ = absorber_type(num_layers=0)
+
+
+@pytest.mark.parametrize(
```

Comment:
now that I look at this potentially this could be one way to go

```py
pytest.mark.parameterize(
    "absorber_type, num_layers",
    [
        (PML, MIN_NUM_PML_LAYERS),
        (StablePML, MIN_NUM_STABLE_PML_LAYERS),
        (Absorber, MIN_NUM_ABSORBER_LAYERS),
    ],
)
def test_num_layers_validator_warning(absorber_type, num_layers):
    """Test the validators in ``PML`` which should display a warning not an error."""
    with AssertLogLevel(None):
        _ = absorber_type(num_layers=num_layers)
    with AssertLogLevel("WARNING"):
        _ = absorber_type(num_layers=num_layers-1)

```